src/psyclone/psyir/nodes/reference.py

```diff
@@ -217,3 +218,70 @@ def reference_accesses(self, var_accesses):
             # The last entry in all_accesses is the one added above
             # in super(Array...). Add the indices to that entry.
             var_info.all_accesses[-1].indices = list_indices
+
+    def is_full_range(self, index):
+        '''Returns True if the specified array index is a Range Node that
+        specified all elements in this index. In the PSyIR this is
+        specified by using LBOUND(name,index) for the lower bound of
+        the range, UBOUND(name,index) for the upper bound of the range
+        and "1" for the range step.
+
+        :param int index: the array index to check.
+
+        :returns: true if the access to this array index is a range \
+            that specifies all index elements. Otherwise returns \
+            false.
+        :rtype: bool
+
+        :raises ValueError: if the supplied index is not less than the \
+            number of dimensions in the array.
+
+        '''
```

Comment:
pylint complains about "too many returns" in this routine but I think the structure is fine so maybe just disable this complaint :-)

Reply:
Disabled.